src/main/scala/uk/gov/hmrc/govukfrontend/views/implicits/RichSelectSupport.scala

```diff
@@ -61,6 +62,22 @@ trait RichSelectSupport {
     def withHeadingAndSectionCaption(heading: Content, sectionCaption: Content): Select =
       withHeadingLabel(select, heading, Some(sectionCaption))((sl, ul) => sl.copy(label = ul))
 
+    def asAccessibleAutocomplete(accessibleAutocomplete: Option[AccessibleAutocomplete] = None): Select = {
+      def toMapOfDataAttributes(accessibleAutocomplete: AccessibleAutocomplete): Map[String, String] =
+        Map(
+          "data-auto-select"     -> accessibleAutocomplete.autoSelect.toString,
+          "data-show-all-values" -> accessibleAutocomplete.showAllValues.toString,
+          "data-default-value"   -> accessibleAutocomplete.defaultValue.getOrElse(""),
+          "data-module"          -> accessibleAutocomplete.dataModule
+        )
+
+      val dataAttributes =
+        toMapOfDataAttributes(accessibleAutocomplete.getOrElse(AccessibleAutocomplete(None))) ++
+          select.attributes
+
+      select.copy(attributes = dataAttributes)
```

Comment:
think this would be slightly clearer as
```suggestion
      val dataAttributes =
        toMapOfDataAttributes(accessibleAutocomplete.getOrElse(AccessibleAutocomplete(None)))

      select.copy(attributes = select.attributes ++ dataAttributes)
```